### PR TITLE
feat: update  latest used time of each layer

### PR DESCRIPTION
**Pull request by Desiki-high**
### Changes:
1. refactor ```adapter.Content```: move to pkg ```content```.
2. update the latest used time of each layer. When fetching any layer, we will update the ```updatedAt``` time in the bolt database if **it exists locally**. The ```updatedAt``` value in bolt database will used by gc.

### CODE REVIEW

File: pkg/content/ported.go [hunk 1]
```diff
@@ -160,7 +162,11 @@ func fetchHandler(ingester content.Ingester, fetcher remotes.Fetcher) images.Han
 			return nil, fmt.Errorf("%v not supported", desc.MediaType)
 		default:
 			_, err, _ := fetchSingleflight.Do(string(desc.Digest), func() (interface{}, error) {
-				return nil, remotes.Fetch(ctx, ingester, fetcher, desc)
+				err := remotes.Fetch(ctx, ingester, fetcher, desc)
```
Comment (imeoer) [inline pkg/content/ported.go:165]: The `err` should be handled first.

File: pkg/content/ported.go [hunk 2]
```diff
@@ -218,7 +224,115 @@ func push(ctx context.Context, store content.Store, pushCtx *containerd.RemoteCo
...
+			indexStack = append(indexStack, desc)
+			m.Unlock()
+			return nil, images.ErrStopHandler
+		default:
+			return nil, nil
+		}
+	})
+
+	pushHandler := remotes.PushHandler(pusher, store)
+	pushHandler = images.Handlers(pushHandler, images.HandlerFunc(func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
```
Comment (imeoer) [inline pkg/content/ported.go:255]: `pushHandler := images.Handlers(remotes.PushHandler(pusher, store), images.HandlerFunc...`

File: pkg/content/ported.go [hunk 3]
```diff
@@ -218,7 +224,115 @@ func push(ctx context.Context, store content.Store, pushCtx *containerd.RemoteCo
...
+		return nil, localContent.UpdateTime(&desc.Digest)
```
Comment (imeoer) [inline pkg/content/ported.go:256]: The updatedat timestamp should be updated once the blob is written to the local disk, instead of pushing stage, right?

Comment (Desiki-high) [reply pkg/content/ported.go:256]: How to know when the converted blob is written? (estargz/nydus)Convert has any hooks?

File: pkg/content/ported.go (hunk 1 above)
Comment (Desiki-high) [reply pkg/content/ported.go:165]: When ```remotes. Fetch``` handled first, if the layer exits locally, the ```err``` will not be ```nil```, so I return ```err``` later to make sure ```UpdateTime``` will be called.

Comment (imeoer) [reply pkg/content/ported.go:165]: Ok, but if the layer does not exist in local and a network error occurs in `remotes.Fetch`, users can't track the real error.

File: pkg/content/ported.go (hunk 3 above)
Comment (imeoer) [reply pkg/content/ported.go:256]: Maybe we can override the method which needs to be hooked like containerd metadata [DB](https://github.com/containerd/containerd/blob/f3a07934b49bf142925a5913e3e19f3528eda0d2/metadata/content.go#L62).

File: pkg/content/ported.go (hunk 1 above)
Comment (Desiki-high) [reply pkg/content/ported.go:165]: Maybe we can check the err type？

Comment (Desiki-high) [reply pkg/content/ported.go:163]: Fixed.

**Review by imeoer — APPROVED** [review]

**Pull request merged**